spec/integration/cursor_reaping_spec.rb

```diff
@@ -32,22 +26,71 @@
       # just the scope, no query is happening
       collection.find.batch_size(2).no_cursor_timeout
 
+      events = EventSubscriber.started_events.select do |event|
+        event.command['killCursors']
+      end
+
       expect(events).to be_empty
     end
 
-    it 'is reaped' do
+    # this let block is a kludge to avoid copy pasting all of this code
+    let(:cursor_id_and_kill_event) do
       expect(Mongo::Operation::KillCursors).to receive(:new).at_least(:once).and_call_original
 
+      cursor_id = nil
+
       # scopes are weird, having this result in a let block
       # makes it not garbage collected
-      2.times { collection.find.batch_size(2).no_cursor_timeout.first }
+      2.times do
+        scope = collection.find.batch_size(2).no_cursor_timeout
+
+        # there is no API for retrieving the cursor
+        scope.each.first
+        # and keep the first cursor
+        cursor_id ||= scope.instance_variable_get('@cursor').id
+      end
+
+      expect(cursor_id).to be_a(Integer)
+      expect(cursor_id > 0).to be true
 
       GC.start
 
       # force periodic executor to run because its frequency is not configurable
       client.cluster.instance_variable_get('@periodic_executor').execute
 
-      expect(events).not_to be_empty
+      started_event = EventSubscriber.started_events.detect do |event|
+        event.command['killCursors'] && event.command['cursors'].include?(cursor_id)
+      end
+
+      expect(started_event).not_to be_nil
+
+      succeeded_event = EventSubscriber.succeeded_events.detect do |event|
+        event.command_name == 'killCursors' && event.request_id == started_event.request_id
+      end
+
+      expect(succeeded_event).not_to be_nil
+
+      # ok is integer on older versions, float on newer versions
+      expect(succeeded_event.reply['ok'].to_i).to be 1
```

Comment:
Not a huge deal, but I think `expect(succeeded_event.reply['ok']).to eq(1)` might be a bit cleaner than casting the type

Reply:
Made the change, thanks.